### PR TITLE
feature:S3C-3532 re-organize and remove the unwanted logic

### PR DESCRIPTION
**Pull request by saitharunthodupunuru**
* Current bean creation logic for VaultAdmin is convoluted and uses too many abstractions. This commit removes unwanted abstractions and simplifies Autowired logic.
* Modified `expirationTime` to `ttl`
* Exception class returns proper HttpStatus now

### CODE REVIEW

File: osis-core/src/test/java/com/scality/osis/service/impl/ScalityOsisServiceTest.java [hunk 1]
```diff
@@ -182,7 +183,7 @@ public void testCreateTenant500(){
 
         when(vaultAdminMock.createAccount(any(CreateAccountRequestDTO.class)))
                 .thenAnswer((Answer<CreateAccountResponseDTO>) invocation -> {
-                    throw new VaultServiceException(500, "EntityAlreadyExists");
+                    throw new VaultServiceException(HttpStatus.INTERNAL_SERVER_ERROR, "EntityAlreadyExists");
```
Comment (anurag4DSB) [inline osis-core/src/test/java/com/scality/osis/service/impl/ScalityOsisServiceTest.java:186]: Internal server error is due to the fault of the server.
Entry already exists is different.
I am confused why we have mix of both?

Here is our exhaustive list for reference: https://github.com/scality/Arsenal/blob/development/8.1/errors/arsenalErrors.json
Similar is followed by AWS.

Comment (anurag4DSB) [reply osis-core/src/test/java/com/scality/osis/service/impl/ScalityOsisServiceTest.java:186]: EntryAlreadyExists is 409
https://github.com/scality/Arsenal/blob/development/8.1/errors/arsenalErrors.json#L394

File: vault-admin-client/src/main/java/com/scality/osis/vaultadmin/impl/VaultAdminImpl.java [hunk 2]
```diff
@@ -217,7 +227,7 @@ public String getAccountsMarker(int offset, String filterKey) throws VaultServic
             cacheListAccountsMarker(listAccountsResponse.getAccounts().size(), marker);
 
           } else{
-            throw new VaultServiceException(400, "Requested offset is outside the total available items");
+            throw new VaultServiceException(HttpStatus.BAD_REQUEST, "Requested offset is outside the total available items");
```
Comment (anurag4DSB) [inline vault-admin-client/src/main/java/com/scality/osis/vaultadmin/impl/VaultAdminImpl.java:230]: Just a suggestion, but I am okay with any.

You can use, InvalidParameterValue: https://github.com/scality/Arsenal/blob/development/8.1/errors/arsenalErrors.json#L426

**Review by anurag4DSB — APPROVED** [review]

**Review by anurag4DSB — APPROVED** [review]
Just a small confusion, otherwise LGTM

File: osis-core/src/test/java/com/scality/osis/service/impl/ScalityOsisServiceTest.java (hunk 1 above)
Comment (saitharunthodupunuru) [reply osis-core/src/test/java/com/scality/osis/service/impl/ScalityOsisServiceTest.java:186]: I will change this anyways in the end, OSIS can return only 201 or 400 for `Create Account`

File: vault-admin-client/src/main/java/com/scality/osis/vaultadmin/impl/VaultAdminImpl.java (hunk 2 above)
Comment (saitharunthodupunuru) [reply vault-admin-client/src/main/java/com/scality/osis/vaultadmin/impl/VaultAdminImpl.java:230]: I am ok to change this but anyways in the end, OSIS can return only 200 for `list Accounts`

**Review by ilkescality — APPROVED** [review]

**Pull request merged**